lib/pagseguro/collection_base.rb

```diff
@@ -0,0 +1,41 @@
+module PagSeguro
+  class CollectionBase
+    extend CollectionManagers::MacroMethods
+    extend Forwardable
+    include Enumerable
+
+    def_delegators :@store, :size, :clear, :empty?, :any?, :each
+
+    def initialize
+      @store = []
+    end
+
+    def collection_manager
+      CollectionManagers::GenericManager
+    end
+
+    def collection_type
+      eval singularize(self.class.to_s)
```

Comment:
@adbatista I open a PR with other suggestion about this whole problem. See please https://github.com/pagseguro/ruby/pull/187